### DISCUSSION TITLE
Tree widget example

### DISCUSSION THREAD

**Discussion by BAISTM** [eclipse-theia/theia#12650] [Q&A] [answered]
Hi,

I start using theia, and I want to know if there is any tutorial or example explaining how to integrate a tree to a view. 

I tried to extend a widget from the `TreeWidget` class but I didn't know what method should I implement and what service should I bind.

Thank you for your help!

**Comment by vince-fugnitto** [comment, marked as answer]
@BAISTM thank you for the discussion, do you mean creating a view which contains an embedded tree? If so you can take a look at the "search in workspace" view which would be an example, albeit maybe a little complicated. The idea is having a view, and attaching the tree as needed to where you want.

- parent view: https://github.com/eclipse-theia/theia/blob/master/packages/search-in-workspace/src/browser/search-in-workspace-widget.tsx
- embedded tree displaying search results: https://github.com/eclipse-theia/theia/blob/master/packages/search-in-workspace/src/browser/search-in-workspace-result-tree-widget.tsx

**Reply by BAISTM** [reply]
Great, thank you 👍
